### ISSUE TITLE
Destination S3: Remove configurable part size

### ISSUE THREAD

**Issue by Phlair** [airbytehq/airbyte#12347]
## Tell us about the problem you're trying to solve
As discussed oncall [here](airbytehq/alpha-beta-issues#68), we feel that the option to allow customers to set the part size in S3 destination is no longer relevant because of the [buffering changes introduced](https://github.com/airbytehq/airbyte/pull/11294) by @ChristopheDuong. We feel that we should remove that option so that people don't end up choosing a higher number like 500MB and causing OOMs.

## Describe the solution you’d like
Remove part size option from spec and ignore it in sync process implementation. This should be possible to implement in a backwards compatible way.

Also if we remove the part size option from S3 spec, we would also have to remove the option from all the destinations that depend on S3 like Redshift, Snowflake, etc.

## Additional context
[Oncall issue with OOMs](airbytehq/alpha-beta-issues#68)

**Comment by tuliren** [comment]
> This should be possible to implement in a backwards compatible way.

Yes. Add `additionalProperties: true` to the spec file should be sufficient.

**Comment by kimerinn** [comment]
Pull request
https://github.com/airbytehq/airbyte/pull/12763

**Comment by kimerinn** [comment]
corect PR
https://github.com/airbytehq/airbyte/pull/13043
